### PR TITLE
Add creator-kit-all.config to build all packages for marduk.

### PR DESCRIPTION
**Pull request by abhijit-mahajani-imgtec**
Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

### CODE REVIEW

**Comment by nikhil-zinjurde-imgtec** [comment]
test ci

**Comment by nikhil-zinjurde-imgtec** [comment]
test ci

**Comment by nikhil-zinjurde-imgtec** [comment]
retest this please

**Comment by nikhil-zinjurde-imgtec** [comment]
ok to test

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
Verifying whether the changes passes Jenkins CI build for normal building of IoT button project (using V=s J=4 parameters.)

The Jenkins CI build with A=1 parameter failed with following errors
````
00:50:16.543 * Security options
00:50:16.543 *
00:50:16.543 Enable access key retention support (KEYS) [Y/n/?] y
00:50:16.543   Enable register of persistent per-UID keyrings (PERSISTENT_KEYRINGS) [N/y/?] n
00:50:16.543   Large payload keys (BIG_KEYS) [N/y/?] n
00:50:16.543   TRUSTED KEYS (TRUSTED_KEYS) [N/m/y/?] n
00:50:16.543   ENCRYPTED KEYS (ENCRYPTED_KEYS) [N/m/y/?] n
00:50:16.543 Restrict unprivileged access to the kernel syslog (SECURITY_DMESG_RESTRICT) [N/y/?] n
00:50:16.543 Enable different security models (SECURITY) [Y/n/?] y
00:50:16.543 Enable the securityfs filesystem (SECURITYFS) [Y/?] y
00:50:16.543 Socket and Networking Security Hooks (SECURITY_NETWORK) [Y/n/?] y
00:50:16.543   XFRM (IPSec) Networking Security Hooks (SECURITY_NETWORK_XFRM) [N/y/?] (NEW) aborted!
00:50:16.543 
00:50:16.543 Console input/output is redirected. Run 'make oldconfig' to update configuration.
00:50:16.543 
00:50:16.544 make[8]: *** [silentoldconfig] Error 1
00:50:16.545 make[7]: *** [silentoldconfig] Error 2
00:50:16.570 
00:50:16.571 The present kernel configuration has modules disabled.
00:50:16.571 Type 'make config' and enable loadable module support.
00:50:16.574 Then build a kernel with module support enabled.
00:50:16.574 
00:50:16.575 make[6]: *** [modules] Error 1
00:50:16.575 make[6]: Leaving directory `/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt/build_dir/target-mipsel_mips32_musl-1.1.11/linux-pistachio_marduk/linux-4.1.13'
00:50:16.576 make[5]: *** [/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt/build_dir/target-mipsel_mips32_musl-1.1.11/linux-pistachio_marduk/linux-4.1.13/.modules] Error 2
00:50:16.577 make[5]: Leaving directory `/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt/target/linux/pistachio'
00:50:16.577 make[4]: *** [compile] Error 2
00:50:16.577 make[4]: Leaving directory `/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt/target/linux'
00:50:16.579 make[3]: *** [target/linux/compile] Error 2
00:50:16.579 make[3]: Leaving directory `/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt'
00:50:16.579 make[2]: *** [/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt/staging_dir/target-mipsel_mips32_musl-1.1.11/stamp/.target_compile] Error 2
00:50:16.580 make[2]: Leaving directory `/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt'
00:50:16.580 make[1]: *** [world] Error 2
00:50:16.580 make[1]: Leaving directory `/home/img-creator/jenkins/workspace/Creator-Build-All/creatorkit/dist/openwrt'
00:50:16.580 make: *** [build_openwrt] Error 2
````

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
test ci

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
test with ca8210 profile

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
Instructions for verifying opkg support:
1.Copy all the build artifacts
OpenWrt-ImageBuilder-pistachio-marduk.Linux-x86_64.tar.bz2
openwrt-pistachio-marduk-marduk-cc2520-rootfs.tar.gz
openwrt-pistachio-marduk-marduk_cc2520-ubifs.img
openwrt-pistachio-marduk-marduk_cc2520-uImage
openwrt-pistachio-marduk-marduk_cc2520-uImage-initramfs
OpenWrt-SDK-pistachio-marduk_gcc-5.2.0_musl-1.1.11.Linux-x86_64.tar.bz2
OpenWrt-Toolchain-pistachio-marduk_gcc-5.2.0_musl-1.1.11.Linux-x86_64.tar.bz2
pistachio_marduk_cc2520.dtb
md5sums
sha256sums
packages.zip
to a webserver to host all these ipks/tarballs/binaries.

So assume its all copied to http://192.168.91.79:8080/
2. Change the download server on openwrt terminal.
root@OpenWrt:/# sed -i "s|http://downloads.openwrt.org/snapshots/trunk/pistachio/marduk/|http://192.168.91.79:8080/|" /etc/opkg/distfeeds.conf

3. root@OpenWrt:/# opkg update
All the packages get pointed to this opkg server and signature verification gets checked.

Check if button-gateway, device-manager, webscripts, awalwm2m are present on the board or not?
4. root@OpenWrt:/# opkg upgrade button-gateway
5. root@OpenWrt:/# opkg remove button-gateway
6. root@OpenWrt:/# opkg install button-gateway

It should install all the dependent packages. If device-manager and button-gateway both needs awalwm2m, then removing button-gateway will not remove awalwm2m. or you cannot remove awalwm2m without removing button-gateway/device-manager. Typical dependency rule applies here too.

7. To install kmod packages, you need to first update your kernel image with this openwrt-pistachio-marduk-marduk_cc2520-uImage present in this webserver, then only any kmod installation/up-gradation is possible.

File: Makefile [hunk 1]
```diff
@@ -14,21 +15,29 @@ openwrt/feeds.conf:
 	cd ../dist/openwrt; \
 	cp ../../build/feeds.conf .; \
 	./scripts/feeds update -a; \
-	./scripts/feeds install -a;
+	./scripts/feeds install -a; \
+	cd feeds/packages; patch -p1 < ../../../../build/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: This should also be protected by build-all?

Comment (abhijit-mahajani-imgtec) [inline Makefile]: Actually it was needed to fix bluez package build issue too...so until we get fix from openwrt packages, we can have this patch to fix those issues

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
Verified

**Pull request merged**